File: python/ql/src/Security/CWE-1427/examples/tool-description-injection.py
from flask import Flask, request
from agents import Agent, FunctionTool, Runner

app = Flask(__name__)


@app.get("/agent")
def agent_route():
    topic = request.args.get("topic")

    # BAD: user input is used in the description of a tool exposed to the agent
    lookup_tool = FunctionTool(
        name="lookup",
        description="Look up reference material about " + topic,
        params_json_schema={},
        on_invoke_tool=lambda ctx, args: "...",
    )

    agent = Agent(
        name="assistant",
        instructions="You are a research assistant that looks up reference material on various topics and answers user questions.",
        tools=[lookup_tool],
    )

    result = Runner.run_sync(agent, request.args.get("message"))

    return result.final_output

File: javascript/ql/src/Security/CWE-1427/examples/tool-description-injection.js
const express = require("express");
const { Agent, tool, run } = require("@openai/agents");

const app = express();

app.get("/agent", async (req, res) => {
    let topic = req.query.topic;

    // BAD: user input is used in the description of a tool exposed to the agent
    const lookupTool = tool({
        name: "lookup",
        description: "Look up reference material about " + topic,
        parameters: {},
        execute: async () => {
            return "...";
        },
    });

    const agent = new Agent({
        name: "assistant",
        instructions: "You are a research assistant that looks up reference material on various topics and answers user questions.",
        tools: [lookupTool],
    });

    const result = await run(agent, req.query.message);

    res.json(result);
});
